src/sentry/integrations/vercel/client.py

```diff
@@ -82,28 +82,32 @@ def create_deploy_webhook(self):
         return response
 
     def get_env_vars(self, vercel_project_id):
-        return self.paginate(self.GET_ENV_VAR_URL % vercel_project_id, "envs")
+        return self.get(self.GET_ENV_VAR_URL % vercel_project_id)
 
     def create_secret(self, vercel_project_id, name, value):
         data = {"name": name, "value": value}
         response = self.post(self.SECRETS_URL, data=data)["uid"]
         return response
 
     def create_env_variable(self, vercel_project_id, key, value):
-        data = {"key": key, "value": value, "target": "production"}
-        response = self.post(self.ENV_VAR_URL % vercel_project_id, data=data)
-        return response
-
-    def update_env_variable(self, vercel_project_id, key, value):
+        data = {"key": key, "value": value, "target": ["production"], "type": "secret"}
```

Comment:
@ceorourke Hmmm so only the `SENTRY_AUTH_TOKEN_ ` environment variables should have the type `secret`. And maybe the commit sha should be `system`. The rest should be `plain` I think.

Reply:
@scefali lemme run through this again on production but I picked secret because in Vercel I went to edit the env vars to see which type they were and I coulda sworn they were all secret.

Reply:
@ceorourke Yea, that's because in the old `v4` endpoint we couldn't actually set a type. But now that we can add a type, we should type them correctly.